### ISSUE TITLE
Import relative when same- or subdirectory

### ISSUE THREAD

**Issue by akomm** [microsoft/vscode#128773]
VScode has already some options for how to auto-import files. 
Feature: Auto-import option which uses relative import, when there is no parent (`..`) traversing, otherwise absolute.

Use-Case: it can reduce path-refactoring needed when moving things around even more than currently possible with baseUrl, paths and VSCode import setting. Take this example:

```ts
// @pathA/a/index.ts
export * from "@pathA/a/item1"
export * from "@pathA/a/item2"
export * from "@pathA/a/item3"
```
In this examples, changes to  `@pathA/a` both in path config, but also when physically moving the directory, would require refactoring of all the pathes, while the following example would not:

```ts
// @pathA/a/index.ts
export * from "./item1"
export * from "./item2"
export * from "./item3"
```

This is sadly not solved by using the VSCode option "take shortest path", as it would also take parent-traversal (`..`) if its shorter. However this makes it messy again to refactor things.

This would also work well with how currently VSCode pick from which point to import. It will always pick the index, which has the shortest way. 

Basically, whenever you would go "up", you would drop to absolute import. Whenever you import "down", it would take relative. That is the idea (not just limited to index, as I'v realized the same benefits apply in general)

**Comment by mjbvz** [comment]
The auto imports feature for JavaScript and TypeScript is powered by the TypeScript project. Please [file a feature request upstream](https://github.com/microsoft/TypeScript/issues/new?template=Feature_request.md)

**Comment by akomm** [comment]
@mjbvz
Also the preference? I thought typescript import options are resolved by typescript, but which are picked is decided by VSC and its setting. I just want to make sure we talk about same thing. Thanks :)
